app/controllers/declaration/EntryIntoDeclarantsRecordsController.scala

```diff
@@ -60,7 +60,9 @@ class EntryIntoDeclarantsRecordsController @Inject()(
   }
 
   private def updateCache(validData: YesNoAnswer)(implicit request: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
-    updateExportsDeclarationSyncDirect(model => model.copy(parties = model.parties.copy(isEntryIntoDeclarantsRecords = Some(validData))))
+    updateExportsDeclarationSyncDirect(
+      model => model.copy(parties = model.parties.copy(isEntryIntoDeclarantsRecords = Some(validData), personPresentingGoodsDetails = None))
```

Comment:
If user come back with back button and choose Yes once again he will loose the data.

I think this putting None inside should be only when User choose No

Reply:
Good job Patryk is on the ball.  I thought that was the case based on the new "user said No" test.

Reply:
The test case is for No, but as I see logic in the controller will always invoke this method, so for Yes we will remove it.

Reply:
I agree with you.